internal/xds/translator/jsonpatch.go

```diff
@@ -84,9 +84,9 @@ func processJSONPatches(tCtx *types.ResourceVersionTable, envoyPatchPolicies []*
 				}
 			}
 
-			// If Path is "" and op is "add", unmarshal and add the patch as a complete
+			// If Path and JSONPath is "" and op is "add", unmarshal and add the patch as a complete
 			// resource
-			if p.Operation.Op == AddOperation && p.Operation.Path == EmptyPath {
+			if p.Operation.Op == AddOperation && p.Operation.IsPathEmpty() && p.Operation.IsJSONPathEmpty() {
```

Comment:
```suggestion
			if p.Operation.Op == AddOperation && (p.Operation.IsPathEmpty() || p.Operation.IsJSONPathEmpty()) {
```

Reply:
I think we need to keep this condition, because else it would always run into that branch if
1. a JsonPath is specified and path is empty
2. a path is specified and JsonPath is empty

Both scenarios will result in a path to a certain element and we don't want to add as a complete resource.

Maybe I didn´t get the intention behind this proposal...?

Reply:
ah got it, thanks for clarifying